build.rs

```diff
@@ -12,4 +12,32 @@ fn main() {
   println!("cargo::rustc-check-cfg=cfg(esp_idf_version_patch, values(\"0\"))");
   println!("cargo::rustc-check-cfg=cfg(esp_idf_version_patch, values(\"1\"))");
   println!("cargo::rustc-check-cfg=cfg(esp_idf_version_patch, values(\"2\"))");
+
+  fn extract_version_component<'a>(
+    mut it: impl Iterator<Item = &'a String>,
+    component_prefix: &str,
+  ) -> usize {
+    it.find_map(|component| {
+      if component.starts_with(component_prefix) {
+        let version_str = component.split_terminator('"').nth(1).unwrap();
+        Some(version_str.parse::<usize>().unwrap())
+      } else {
+        None
+      }
+    })
+    .unwrap()
+  }
+
+  let esp_idf_info = embuild::espidf::sysenv::cfg_args().unwrap();
+  let esp_idf_major = extract_version_component(esp_idf_info.args.iter(), "esp_idf_version_major");
+  let esp_idf_minor = extract_version_component(esp_idf_info.args.iter(), "esp_idf_version_minor");
+  let esp_idf_patch = extract_version_component(esp_idf_info.args.iter(), "esp_idf_version_patch");
+
+  println!("cargo::rustc-check-cfg=cfg(cpfd)");
+  if esp_idf_major > 5
+    || (esp_idf_major == 5 && esp_idf_minor > 2)
+    || (esp_idf_major == 5 && esp_idf_minor == 2 && esp_idf_patch > 0)
+  {
+    println!("cargo::rustc-cfg=cpfd");
+  }
```

Comment:
This can be simplified as follows
```rust
if (esp_idf_major, esp_idf_minor, esp_idf_patch) > (5, 2, 0) {
  println!("cargo::rustc-cfg=cpfd");
}
```

Reply:
Nice! TIL
I updated the patch a bit and used that feature of `Vec`. Let me know what you think.